src/porepy/grids/md_grid.py

```diff
@@ -439,120 +413,86 @@ def remove_subdomain(self, sd: pp.Grid) -> None:
-
-        Intended for keeping the subdomain ordering after removing a subdomain from the
-        mixed-dimensional grid. In this way, the interface sorting will not be disturbed
-        by the removal, but no gaps in are created.
+        Sorting by descending interface dimension and increasing MortarGrid id,
+        see argsort_grids.
 
         Args:
-            removed_number (int): node_number of the removed grid.
+            interfaces: List of interfaces.
 
-        """
+        Returns:
+            List containing the same subdomains, sorted.
 
-        # Loop over grids in decreasing dimensions
-        for dim in range(self.dim_max(), self.dim_min() - 1, -1):
-            for sd, data in self.subdomains(return_data=True, dim=dim):
-                if "node_number" not in self._subdomain_data[sd]:
-                    # It is not clear how severe this case is. For the moment,
-                    # we give a warning, and hope the user knows what to do
-                    warnings.warn(
-                        "Tried to update subdomain ordering where none exists"
-                    )
-                    # No point in continuing with this subdomain.
-                    continue
-
-                # Obtain the old node number
-                data = self._subdomain_data[sd]
-                old_number = data.get("node_number", -1)
-                # And replace it if it is higher than the removed one
-                if old_number > removed_number:
-                    data["node_number"] = old_number - 1
+        """
+        inds = self.argsort_grids(interfaces)
+        sorted_interfaces = [interfaces[i] for i in inds]
+        return sorted_interfaces
 
-    def sort_subdomains(self, subdomains: List[pp.Grid]) -> List[pp.Grid]:
-        """Sort subdomains according to the subdomain ordering (attribute 'node_number'
-        in the data dictionary associated with the subdomain).
+    def argsort_grids(
+        self, grids: Iterable[Union[pp.Grid, pp.MortarGrid]]
+    ) -> np.ndarray:
+        """Return indices that would sort the subdomains or interfaces.
 
+        Sorting is done according to two criteria:
+            Descending dimension
+            Ascending (Mortar)Grid id (for each dimension). The id is assigned on instantiation
+            of a (mortar) grid. The codimension is not considered for interfaces.
         Args:
-            subdomains: List of subdomains.
+            grids: List of subdomain grids.
 
         Returns:
-            sorted_subdomain_data: The same subdomains, sorted.
-
+            np.ndarray that sorts the grids, e.g.
+                inds = mdg.argsort_subdomains(subdomain_list)
+                sorted_subdomains = [subdomain_list[i] for i in inds]
         """
-        assert all(["node_number" in self._subdomain_data[sd] for sd in subdomains])
-
-        return sorted(subdomains, key=lambda n: self.node_number(n))
+        sort_inds_all: np.ndarray = np.array([], dtype=int)
```

Comment:
Generally, I prefer to build such arrays by making a list of numpy arrays and then to a hstack at the end. This will be more efficient, but it may not be worth the effort to change it here.

Reply:
Would it be list of arrays or list of ints in this case? I'm inclined to think we should care about efficiency, as this method may be called relatively often.

Reply:
List of arrays. The time will be saved generating many small and one big array, instead of progressively larger arrays.